dataservice/extensions.py

```diff
@@ -2,6 +2,59 @@
 from flask_sqlalchemy import SQLAlchemy, Model
 
 
-db = SQLAlchemy()
+class CRUDMixin(Model):
+    """
+    Mixin that adds convenience methods for CRUD
+    (create, read, update, delete) operations.
+    """
+
+    @classmethod
+    def create(cls, **kwargs):
+        """
+        Create a new record and save it the database.
+        """
+        instance = cls(**kwargs)
+        return instance.save()
```

Comment:
What if your creating this object with a bunch of other objects?
Eg: I create a person with a demographic.
I probably would want to make one single transaction rather than two.

Reply:
You can still do that. See tests for examples (in branch demographic-entity: tests > demographic > test_demographic_models)

Here's an example for convenience too:
```python
# Create demographic
        demo_id = 'demo_1'
        data = {
            'external_id': demo_id,
            'race': 'asian',
            'ethnicity': 'not hispanic',
            'gender': 'female',
            'sex': 'F'
        }
        d = Demographic(**data)

        # Create and save person with demographic
        person_id = 'Test subject 0'
        data = {
            'external_id': person_id,
            'demographic': d
        }
        p = Person.create(**data)
```
Is this what you're asking about?

Reply:
Yes, however, this is done in two transactions. It would be best practice to have these done in one atomic operation.
I see that you can flag whether to commit or not in the `save` method, but I'm of the opinion that this commit should be done explicitly at the end of the transaction (the request), at least for simple crud operations (batch is another topic).